### PR TITLE
Update ODIAC data content 

### PR DESCRIPTION
**Pull request by paridhi-parajuli**
## Why are you creating this Pull Request?

- Added minor content changes for [ODIAC](https://docs.google.com/document/d/1NLAyzK3BXrFK9T8eXI8KCw_QDevDQ4S5kf4Az0tUX28/edit)
- Changed the stac collection id to `odiac-ffco2-monthgrid-v2023`

### CODE REVIEW

**Comment by paridhi-parajuli** [comment]
For now he exploration wont show up because the collection `odiac-ffco2-monthgrid-v2023` is only in the dev stac. For the exploration to work locally, change the api endpoints to `https://dev.ghg.center/api/*`

File: datasets/odiac-ffco2-monthgrid-v2023.data.mdx [hunk 1]
```diff
@@ -118,30 +118,35 @@ layers:
...
   </Prose>
 </Block>
 <Block>
   <Prose>
     ## Source Data Product Citation
     Oda, Tomohiro., Maksyutov, Shamil. 2015. ODIAC Fossil Fuel CO₂ Emissions Dataset (Version name: ODIAC2022), Center for Global Environmental Research, National Institute for Environmental Studies, https://doi.org/10.17595/20170411.001. (Reference date: 2023/06/14)
```
Comment (Jeanne-le-Roux) [inline datasets/odiac-ffco2-monthgrid-v2023.data.mdx:135]: Update the citation to the following:

Tomohiro Oda, Shamil Maksyutov, (2015). ODIAC Fossil Fuel CO2 Emissions Dataset (Version name: ODIAC2023), Center for Global Environmental Research, National Institute for Environmental Studies, [DOI:10.17595/20170411.001](http://doi.org/10.17595/20170411.001). (Reference date: 2024/07/15)

**Review by Jeanne-le-Roux — CHANGES_REQUESTED** [review]

**Comment by slesaad** [comment]
fix merge conflict and mark ready for review

**Review by slesaad — APPROVED** [review]

File: datasets/odiac-ffco2-monthgrid-v2023.data.mdx [hunk 2]
```diff
@@ -116,21 +116,26 @@ layers:
...
   </Prose>
 </Block>
 <Block>
   <Prose>
     ## Source Data Product Citation
-    Oda, Tomohiro., Maksyutov, Shamil. 2015. ODIAC Fossil Fuel CO₂ Emissions Dataset (Version name: ODIAC2022), Center for Global Environmental Research, National Institute for Environmental Studies, https://doi.org/10.17595/20170411.001. (Reference date: 2023/06/14)
+    Tomohiro Oda, Shamil Maksyutov, (2015). ODIAC Fossil Fuel CO2 Emissions Dataset (Version name: ODIAC2023), Center for Global Environmental Research, National Institute for Environmental Studies, DOI:10.17595/20170411.001. (Reference date: 2024/07/15)
```
Comment (Jeanne-le-Roux) [inline datasets/odiac-ffco2-monthgrid-v2023.data.mdx:133]: Hyperlink DOI:10.17595/20170411.001 with http://doi.org/10.17595/20170411.001

File: datasets/odiac-ffco2-monthgrid-v2023.data.mdx [hunk 3]
```diff
@@ -116,21 +116,26 @@ layers:
...
+
+    ## Version History
+    The current dataset version is v2023, which replaced the v2022 dataset in the US GHG Center in June 2024. Summary of dataset update:
```
Comment (Jeanne-le-Roux) [inline datasets/odiac-ffco2-monthgrid-v2023.data.mdx:136]: Change June to August

File: datasets/odiac-ffco2-monthgrid-v2023.data.mdx [hunk 4]
```diff
@@ -116,21 +116,26 @@ layers:
...
+    - The year 2023 version of ODIAC is based on the latest country fossil fuel CO2 estimates (2000-2020) from the CDIAC team at the Appalachian State University, which includes an additional year of data compared to v2022.
```
Comment (Jeanne-le-Roux) [inline datasets/odiac-ffco2-monthgrid-v2023.data.mdx:137]: Add subscript for CO2

File: datasets/odiac-ffco2-monthgrid-v2023.data.mdx (hunk 2 above)
Comment (Jeanne-le-Roux) [inline datasets/odiac-ffco2-monthgrid-v2023.data.mdx:133]: Add subscript for CO2

**Review by Jeanne-le-Roux — CHANGES_REQUESTED** [review]
Also ensure the Access Data > Data Browser link is updated (in the preview link it still points to v2022 data).

**Review by Jeanne-le-Roux — APPROVED** [review]

**Pull request merged**